File: src/AutomatonTheory/JGraph/DialogBoxes/RemoveStateDialog.java
package AutomatonTheory.JGraph.DialogBoxes;

import javax.swing.*;

public class RemoveStateDialog {
    public String[] States;
    public JComboBox<String> removeStateComboBox = new JComboBox<String>();
    public String stateToRemove = "";

    public int valor;
    public RemoveStateDialog(String[] states){
        States = states;
    }

    public void displayGUI() {
        valor = JOptionPane.showConfirmDialog(null,
                getPanel(),
                "Set Initial State",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.PLAIN_MESSAGE);
        stateToRemove = removeStateComboBox.getSelectedItem().toString();
    }

    private JPanel getPanel(){

        JPanel panel = new JPanel();
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
        JLabel originLabel = new JLabel("Choose State to Remove: ");
        removeStateComboBox = new JComboBox<String>(States);

        panel.add(originLabel);
        panel.add(removeStateComboBox);

        return panel;
    }

    public static void main(String[] args){
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                new RemoveStateDialog(new String[]{"q0","q1"}).displayGUI();
            }
        });
    }
}

File: src/AutomatonsTheory/Swing/DialogBoxes/RemoveStateDialog.kt
package AutomatonsTheory.Swing.DialogBoxes

import javax.swing.*

class RemoveStateDialog(var States: Array<String>) {
    var removeStateComboBox = JComboBox<String>()
    var stateToRemove = ""

    var valor: Int = 0

    fun displayGUI() {
        valor = JOptionPane.showConfirmDialog(null,
                panel,
                "Set Initial State",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.PLAIN_MESSAGE)
        stateToRemove = removeStateComboBox.selectedItem.toString()
    }

    private val panel: JPanel
        get() {

            val panel = JPanel()
            panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
            val originLabel = JLabel("Choose State to Remove: ")
            removeStateComboBox = JComboBox(States)

            panel.add(originLabel)
            panel.add(removeStateComboBox)

            return panel
        }

    companion object {

        @JvmStatic fun main(args: Array<String>) {
            SwingUtilities.invokeLater { RemoveStateDialog(arrayOf("q0", "q1")).displayGUI() }
        }
    }
}
